### DISCUSSION TITLE
Web app loding problem

### DISCUSSION THREAD

**Discussion by akashdip2001** [Schweinepriester/achievement-testarea#51] [Q&A]
I created a Web app for my website with the helping of ai. Everything is good but one error is there .

Inside my web many videos are attached with youtube api.

The videos play very well inside the app. And also open in full screen with automatically rotating screen.

But when I exit from full screen after watching any video - a **While Screen problem** appears.

| [Download the App](https://github.com/Engineering-college-btech/custom-video-player/releases/tag/White_Screen_Problem) |
|---

Website - which runs inside the app
| [Website source code](https://github.com/Engineering-college-btech/custom-video-player) |
|---

App password - subscribe_akashdip

The problem video -
https://github.com/drknzz/GitHub-Achievements/assets/81384987/4394b9d9-0de9-4824-a65a-3ed7deae38f2

| [The video on YouTube](https://youtube.com/shorts/n0SjtETHJFE?feature=share) |
|---

**Comment by giangeralcus** [comment]
That white screen after exiting fullscreen is a classic Android WebView `WebChromeClient` bug: when the custom view is dismissed, the `WebView` isn't re-attached to its original container properly, so the surface ends up blank.

Check these, in order of likelihood:

1. **Activity recreation on rotation.** You said fullscreen auto-rotates the screen. When the activity is destroyed/recreated on rotate, the WebView is re-inflated and shows white/blank. Add this to your activity in `AndroidManifest.xml`:
   `android:configChanges="orientation|screenSize|keyboardHidden"`
   so rotation doesn't recreate the activity. This alone fixes most of these cases.
2. **`onHideCustomView` must restore state.** In your `WebChromeClient`, `onHideCustomView()` should: remove the fullscreen view from the decor view, re-add the original `WebView` (the reference you saved in `onShowCustomView`) to its original parent, and stop any fullscreen flags. If you skip re-attaching the original view, you get exactly a white screen.
3. After restoring, call `webView.requestLayout()` and `webView.invalidate()` to force a redraw.
4. Make sure you never call `webView.destroy()` or `reload()` during the fullscreen transition — that drops the surface.

If you share the relevant `WebChromeClient` code I can point at the exact line.
